server/__init__.py

```diff
@@ -60,9 +60,9 @@ def get_send_file_max_age(self, name):
 
 @app.before_request
 def redirect_www():
-    """Redirect www and beta requests to bare domain."""
+    """Redirect subdomain requests to bare domain."""
     urlparts = urlparse(request.url)
-    if urlparts.netloc == 'www.httparchive.org' or urlparts.netloc == 'beta.httparchive.org':
+    if urlparts.netloc == 'www.httparchive.org' or urlparts.netloc == 'beta.httparchive.org' or urlparts.netloc == 'legacy.httparchive.org':
```

Comment:
```suggestion
    if (
        urlparts.netloc == "www.httparchive.org"
        or urlparts.netloc == "beta.httparchive.org"
        or urlparts.netloc == "legacy.httparchive.org"
    ):
```

Let's also add a test for this.